tests/api-watch-diff/test.ml

```diff
@@ -1,119 +1,16 @@
-        type_manifest = Some Predef.type_int;
-        type_variance = [];
-        type_separability = [];
-        type_is_newtype = false;
-        type_expansion_scope = 0;
-        type_loc = Location.none;
-        type_attributes = [];
-        type_immediate = Unknown;
-        type_unboxed_default = false;
-        type_uid = Uid.internal_not_actually_unique;
-      },
-      Trec_not,
-      Exported )
-
-(* Signature for type t:
-     > type unused_type = string *)
-
-let unused_type_sig =
-  Sig_type
-    ( Ident.create_persistent "unused_type",
-      {
-        type_params = [];
-        type_arity = 0;
-        type_kind = Type_abstract;
-        type_private = Public;
-        type_manifest = Some Predef.type_string;
-        type_variance = [];
-        type_separability = [];
-        type_is_newtype = false;
-        type_expansion_scope = 0;
-        type_loc = Location.none;
-        type_attributes = [];
-        type_immediate = Unknown;
-        type_unboxed_default = false;
-        type_uid = Uid.internal_not_actually_unique;
-      },
-      Trec_not,
-      Exported )
-
-(* Signature for value f:
-    > val f : t -> string *)
-
-let val_f_sig =
-  Sig_value
-    ( Ident.create_persistent "f",
-      {
-        val_type =
-          create_expr
-            (Tarrow
-               ( Nolabel,
-                 create_expr
-                   (Tconstr
-                      (Path.Pident (Ident.create_persistent "t"), [], ref Mnil))
-                   ~level:0 ~scope:0 ~id:2,
-                 Predef.type_string,
-                 commu_ok ))
-            ~level:0 ~scope:0 ~id:4;
-        val_kind = Val_reg;
-        val_loc = Location.none;
-        val_attributes = [];
-        val_uid = Uid.internal_not_actually_unique;
-      },
-      Exported )
-
-(* Signature for value g:
-    > val g : t -> t *)
-
-let val_g_sig =
-  Sig_value
-    ( Ident.create_persistent "g",
-      {
-        val_type =
-          create_expr
-            (Tarrow
-               ( Nolabel,
-                 create_expr
-                   (Tconstr
-                      (Path.Pident (Ident.create_persistent "t"), [], ref Mnil))
-                   ~level:0 ~scope:0 ~id:5,
-                 create_expr
-                   (Tconstr
-                      (Path.Pident (Ident.create_persistent "t"), [], ref Mnil))
-                   ~level:0 ~scope:0 ~id:6,
-                 commu_ok ))
-            ~level:0 ~scope:0 ~id:7;
-        val_kind = Val_reg;
-        val_loc = Location.none;
-        val_attributes = [];
-        val_uid = Uid.internal_not_actually_unique;
-      },
-      Exported )
-
-(* Signature for file ref.mli:
+(* Signature for ref.mli:
       > type t = int
       > type unused_type = string
       > val f : t -> string *)
 
-let ref_signature = [ t_sig; unused_type_sig; val_f_sig ]
+let ref_signature =
+  Test_helpers.compile_interface
+    "type t = int\ntype unused_type = string\nval f : t -> string"
```

Comment:
since we're going to write OCaml code in those string literals I'd recommend using the `{|` and `|}` string delimiters.

That will save some annoying escaping and allow us to have the string litterals look like actual code:
```suggestion
    {|
    type t = int
    type unused_type = string
    val f : t -> string
    |}
```

Reply:
Same applies to other occurences of course!

No need to use `\n` either. An actual newline should work!